paper-tab.d.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * DO NOT EDIT
+ *
+ * This file was automatically generated by
+ *   https://github.com/Polymer/gen-typescript-declarations
+ *
+ * To modify these typings, edit the source file(s):
+ *   paper-tab.html
+ */
+
+/// <reference path="../polymer/types/polymer.d.ts" />
+/// <reference path="../iron-behaviors/iron-button-state.d.ts" />
+/// <reference path="../iron-behaviors/iron-control-state.d.ts" />
+/// <reference path="../iron-flex-layout/iron-flex-layout.d.ts" />
+/// <reference path="../paper-behaviors/paper-ripple-behavior.d.ts" />
+
+/**
+ * `paper-tab` is styled to look like a tab.  It should be used in conjunction with
+ * `paper-tabs`.
+ *
+ * Example:
+ *
+ *     <paper-tabs selected="0">
+ *       <paper-tab>TAB 1</paper-tab>
+ *       <paper-tab>TAB 2</paper-tab>
+ *       <paper-tab>TAB 3</paper-tab>
+ *     </paper-tabs>
+ *
+ * ### Styling
+ *
+ * The following custom properties and mixins are available for styling:
+ *
+ * Custom property | Description | Default
+ * ----------------|-------------|----------
+ * `--paper-tab-ink` | Ink color | `--paper-yellow-a100`
+ * `--paper-tab` | Mixin applied to the tab | `{}`
+ * `--paper-tab-content` | Mixin applied to the tab content | `{}`
+ * `--paper-tab-content-unselected` | Mixin applied to the tab content when the tab is not selected | `{}`
+ *
+ * This element applies the mixin `--paper-font-common-base` but does not import `paper-styles/typography.html`.
+ * In order to apply the `Roboto` font to this element, make sure you've imported `paper-styles/typography.html`.
+ */
+interface PaperTabElement extends Polymer.Element, Polymer.IronControlState, Polymer.IronButtonState, Polymer.PaperRippleBehavior {
+
+  /**
+   * If true, the tab will forward keyboard clicks (enter/space) to
+   * the first anchor element found in its descendants
+   */
+  link: boolean|null|undefined;
+  hostAttributes: object|null;
```

Comment:
Is `listeners` missing here because it's somehow marked as special and removed? If so, `hostAttributes` should probably be handled the same way.

Reply:
Also, why `|null`?

Reply:
`listeners` is handled special by the Polymer element scanner in Analyzer. It just knows not to emit it as a property. It has no special handling for `hostAttributes`. For `hostAttributes` I just added a `@private` annotation, which is what we've been doing most of the other times we notice it.

I'm not sure why Analyzer adds `|null` there, it's a little odd. Since there's no type annotation, it's doing something based purely on the JS syntax it sees. Unfortunately there are lots of little oddities like this, and I've kind of reached the point where I'm trying not to let perfect be the enemy of the good, and just get these typings out there for people to use, knowing that as we make Analyzer smarter, the typings will automatically improve too (there's a tedium pass to re-generate across all the repos).